onnxruntime/core/providers/cuda/math/topk.cc

```diff
@@ -95,9 +95,9 @@ Status TopK<inputk>::ComputeInternal(OpKernelContext* ctx) const {
   if (IS_PRIM_TYPE(int16_t)) return TOPKIMPL(int16_t);
   if (IS_PRIM_TYPE(int32_t)) return TOPKIMPL(int32_t);
   if (IS_PRIM_TYPE(int64_t)) return TOPKIMPL(int64_t);
+  if (IS_PRIM_TYPE(MLFloat16)) return TOPKIMPL(MLFloat16);
```

Comment:
nits: 
1) Maybe the floating types can come ahead of the integer types in the conditional branching order ? 
2) A test would be nice for the new type supported.